src/coreclr/jit/valuenum.cpp

```diff
@@ -7167,6 +7208,43 @@ ValueNum ValueNumStore::EvalHWIntrinsicFunBinary(var_types      type,
 
         switch (ni)
         {
+#ifdef TARGET_ARM64
+            case NI_Vector64_op_Equality:
+            case NI_Vector128_op_Equality:
+            case NI_Vector64_EqualsAll:
+            case NI_Vector128_EqualsAll:
+#else
+            case NI_Vector128_op_Equality:
+            case NI_Vector256_op_Equality:
+            case NI_Vector512_op_Equality:
+            case NI_Vector128_EqualsAll:
+            case NI_Vector256_EqualsAll:
+            case NI_Vector512_EqualsAll:
+#endif
+            {
+                if (!varTypeIsFloating(baseType))
+                {
+                    return EvaluateBinarySimd(this, GT_EQ, /* scalar */ false, type, baseType, arg0VN, arg1VN);
+                }
+                break;
+            }
+
+#ifdef TARGET_ARM64
+            case NI_Vector64_op_Inequality:
+            case NI_Vector128_op_Inequality:
+#else
+            case NI_Vector128_op_Inequality:
+            case NI_Vector256_op_Inequality:
+            case NI_Vector512_op_Inequality:
+#endif
+            {
+                if (!varTypeIsFloating(baseType))
+                {
+                    return EvaluateBinarySimd(this, GT_NE, /* scalar */ false, type, baseType, arg0VN, arg1VN);
+                }
+                break;
+            }
```

Comment:
Do we want to handle `GT`, `LT`, `GE`, and `LE` simultaneously?

Likewise given you've added the support for computing the vector version in order to make `bool` work, should we just handle the intrinsics that produce a vector as well?

Reply:
Those had no hits so I just didn't want to add more code and tests 🙂 Although, even EQ/NE don't have hits, I just found a use case outside.

Reply:
I think this is the type of scenario where it’s a core comparison on SIMD and so we want to generally handle it even if our own code doesn’t have any hits today.

it’s odd to cover some of the relational comparisons and not the others